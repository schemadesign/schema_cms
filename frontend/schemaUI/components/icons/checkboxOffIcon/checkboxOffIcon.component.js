import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import CheckboxOffSVG from '../../../images/icons/checkbox-off.svg';
import { getStyles } from './checkboxOffIcon.styles';
import { withStyles } from '../../styles/withStyles';

export class CheckboxOffIconComponent extends PureComponent {
  static propTypes = {
    customStyles: PropTypes.object,
  };

  render() {
    const { customStyles, theme, ...restProps } = this.props;
    const containerStyles = getStyles(theme);
    const styles = { ...containerStyles, ...customStyles };

    return <CheckboxOffSVG {...restProps} style={styles} />;
  }
}

export const CheckboxOffIcon = withStyles(CheckboxOffIconComponent);
