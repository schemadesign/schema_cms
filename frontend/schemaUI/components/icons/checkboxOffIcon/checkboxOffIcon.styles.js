import { primary } from '../../../utils/theme';

export const containerStyles = {
  stroke: primary.iconDark,
  width: '30px',
  height: '30px',
};
