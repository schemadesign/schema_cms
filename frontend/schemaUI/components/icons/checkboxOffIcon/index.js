export { CheckboxOffIcon } from './checkboxOffIcon.component';
