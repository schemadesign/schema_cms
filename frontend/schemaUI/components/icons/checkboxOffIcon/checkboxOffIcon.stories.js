import React from 'react';
import { storiesOf } from '@storybook/react';

import { CheckboxOffIcon } from './checkboxOffIcon.component';
import { withTheme } from '../../../.storybook/decorators';

const defaultProps = { customStyles: { stroke: 'blue' } };

storiesOf('Icons/CheckboxOffIcon', module)
  .addDecorator(withTheme())
  .add('Default', () => <CheckboxOffIcon />)
  .add('with custom styles', () => <CheckboxOffIcon {...defaultProps} />);
