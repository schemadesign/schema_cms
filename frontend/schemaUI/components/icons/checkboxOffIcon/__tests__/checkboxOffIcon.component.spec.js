import React from 'react';
import { shallow } from 'enzyme';

import { CheckboxOffIconComponent } from '../checkboxOffIcon.component';

describe('CheckboxOffIcon: Component', () => {
  const defaultProps = {};

  const component = props => <CheckboxOffIconComponent {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });
});
