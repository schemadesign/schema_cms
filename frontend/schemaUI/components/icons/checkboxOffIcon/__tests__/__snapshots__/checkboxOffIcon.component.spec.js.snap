// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxOffIcon: Component should render correctly 1`] = `
<checkbox-off.svg
  style={
    Object {
      "height": "30px",
      "stroke": "#1d1d20",
      "width": "30px",
    }
  }
/>
`;
